### ISSUE TITLE
GFF format trouble 

### ISSUE THREAD

**Issue by mariemh23** [jaumlrc/ProphET#20]
Hello, 
I have trouble in using prophET. I validated my GFF file and  i still receive this error "The following line does not have 9 columns separated by <tab>, a GFF format requirement.
Check if the columns are seperated by multiple <spaces> instead". 
Any help ?
Thanks

**Comment by scottdaniel** [comment]
Could you attach the gff file so we could take a look? Save it as raw text.

**Comment by mariemh23** [comment]
[
[](url)
](url)

**Comment by mariemh23** [comment]
Here is my GGF3 file. I have removed the downstream fasta sequences. Thanks
[
[GGF3_ref.txt](https://github.com/jaumlrc/ProphET/files/1414287/GGF3_ref.txt)
](url)

**Comment by scottdaniel** [comment]
Hi,

Well, I read the table into R and it appears like there are no extra columns:
`thing=read.delim("GGF3_ref.txt",comment.char = "#",header = F)`

Any help developers?

**Comment by gkapatai** [comment]
I had exactly the same problem that was solved when I removed the  downstream fasta. However, I got stuck in the next step when the GFFfile.pm script requires Parent in the final column. Obviously this is not provided by Prokka so my question is can I comment out the search for Parent or is it required in downstream steps? If so, what tools can I use to get the annotation format ProPHET requires?

**Comment by scottdaniel** [comment]
@gkapatai I'm getting this same problem too with the complaint about the "Parent". I've tried genomes and gffs from both https://www.patricbrc.org/ and ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
I wonder where they are getting their gff's from that actually work?

NB: If they are needed downstream, I could whip together a script to add "Parent" lines that exactly matched to CDS and see if that worked. But until I hear back from the developers I have other things to do.

**Comment by gustavo11** [comment]
Sorry @gkapatai @scottdaniel  and @mariemh23 . I was in the process of changing jobs... and it seems that my github email was not updated. I'm seeing this request only today. I will resolve this issue in the next couple of days.

**Comment by mariemh23** [comment]
Any news ?

**Comment by gustavo11** [comment]
Yes. I prepared a GFF converter. I will place it on the repository. Tonight. Thanks for you patience.

**Comment by mariemh23** [comment]
forgive me for insisting, but the problem remains unresolved. Could you tell us what software you used to generate the gff files?

**Comment by gustavo11** [comment]
Dear Mariem.
The GFF should follow the format defined by the "The Sequence Ontology Consortium", please see section
"Before running ProphET in your favorite bacterial genome" in the documentation. 

But please bear, I will upload a converter for your GFF today.

**Comment by mariemh23** [comment]
Thank you

**Comment by gustavo11** [comment]
@mariemh23 The converter (gff_rewrite.pl) was uploaded as part of GFFLIb package.
https://github.com/gustavo11/GFFLib

This package is automatically downloaded when Prophet is installed. But fell free download it separately from the URL above.

usage:
gff_rewrite.pl --input <input GFF> \
--output <output GFF>  \
--add_missing_features

Please do not hesitate to contact me with any issue.

**Comment by mariemh23** [comment]
Thank you for your assistance
Unfortunately the installation does not work anymore therefore the folder Phage_proteins_without_ABC-t.db  is no longer generated. I got this error
Downloading GFFLib ...
svn: E670002: Unable to connect to a repository at URL 'https://github.com/gustavo11/GFFLib/trunk'
svn: E670002: Name or service not known
ERROR: Unable to download GFFLib from github

**Comment by gustavo11** [comment]
Hi @mariemh23. Please try again. I have change the method of downloading GFFLib. The installation works fine in our machines, but it seems that Subversion version in your machine cannot reach that URL.

**Comment by gustavo11** [comment]
Hi @mariemh23. Is it working for you now? Thank you very much for the feedback.

**Comment by gustavo11** [comment]
Dear @scottdaniel. I have allocated some time every week to maintain and update ProphET. Please tell me if you need anything.

**Comment by mariemh23** [comment]
@gustavo11 I installed it again and when I try again to test it on the test data, I get the same error message.

cannot stat ‘/home/hanachi/softwares/ProphET/PhrophET_phage_proteins_database.dir/phage_db.summary.stats’: No such file or directory

**Comment by gustavo11** [comment]
Hi @mariemh23, Would you please do me a favor?
Can you please download and install it again in a different directory? Please execute INSTALL.pl from ProphET's home directory.

If the INSTALL.pl fails again would you please execute it one more time as shown below, this will redirect stdout and stderr to a file:
./INSTALL.pl  &> ProphET.stderr.stdout

and then issue the following command to list the content of ProphET home directory:
ls -R > ProphET.home.txt

Finally, please send me both files: ProphET.stderr.stdout and ProphET.home.txt
This will help know more about the issue that you are having so we can quickly address it.

Thank you very much mariemh23.

**Comment by mariemh23** [comment]
Yes, finally when I reloaded again, it worked on the test data. I'm trying them on my own data after converting the gff file. I'll keep you posted.

**Comment by mariemh23** [comment]
Thanks @gustavo11  for your assistance. It finally works
